kernel/aster-nix/src/process/sync/condvar.rs

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::{collections::LinkedList, sync::Arc};
+use core::{
+    sync::atomic::{AtomicBool, Ordering},
+    time::Duration,
+};
+
+use aster_frame::sync::{guard_lock, MutexGuard, SpinLock};
+
+use super::Pauser;
+use crate::Errno;
+
+/// The lock error type for situations where the condvar is not notified.
+pub enum LockErr<Guard> {
+    Timeout(Guard),
+    Interrupted(Guard),
+    Unknown(Guard),
+}
+/// LockResult, different from Rust std.
+/// The result of a lock operation.
+pub type LockResult<Guard> = Result<Guard, LockErr<Guard>>;
+
+impl<Guard> LockErr<Guard> {
+    pub fn unwrap(self) -> Guard {
+        match self {
+            LockErr::Timeout(guard) => guard,
+            LockErr::Interrupted(guard) => guard,
+            LockErr::Unknown(guard) => guard,
```

Comment:
This method can be renamed as `into_inner`, since it consumes the struct and returns the inner guard.

Further, the match statement can be simplified as
```rust
match self {
    LockErr::Timeout(guard) | LockErr::Interrupted(guard) | LockErr::Unknown(guard)=> guard 
}

```

Reply:
Your expression looks better!